### PR TITLE
[PM-7720] Add android apk package name support to passkey-rs

### PR DESCRIPTION
**Pull request by coroiu**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+## Paskey v0.3.1
+### passkey-client v0.3.1
+
+The client now supports additional user-defined properties in the client data, while also clarifying how the client
+handles client data and its hash.
+
+- Change `register` and `authenticate` to take a `ClientData<E>` instead of `Option<Vec<u8>>`.
+- Custom client data hashes are now specified using `DefaultClientDataWithCustomHash(Vec<u8>)` instead of 
+  `Some(Vec<u8>)`.
+- Additional fields can be aedded to the client data using `DefaultClientDataWithExtra(ExtraData)`.
```
Comment (dani-garcia) [inline CHANGELOG.md]:
```suggestion
- Additional fields can be added to the client data using `DefaultClientDataWithExtra(ExtraData)`.
```

File: passkey-types/src/webauthn/attestation.rs [hunk 2]
```diff
@@ -603,6 +606,11 @@ pub struct CollectedClientData {
     #[serde(default, serialize_with = "truthiness")]
     pub cross_origin: Option<bool>,
 
+    /// CollectedClientData can be extended by the user of this library, this accounts for
+    /// keys that are unknown to the library, but may be known to the user.
+    #[serde(flatten)]
```
Comment (dani-garcia) [inline passkey-types/src/webauthn/attestation.rs:611]: Did you end up testing if having two `serde(flatten)` was okay for serialization/deserialization? Might be a good idea to add some unknown_keys to some of the tests to make sure they survive.

**Review by dani-garcia — APPROVED** [review]
LGTM, just a typo and a question about the serde flatten thing.

File: passkey-types/src/webauthn/attestation.rs (hunk 2 above)
Comment (coroiu) [reply passkey-types/src/webauthn/attestation.rs:611]: Ah yes, good point, I wanted to add tests for it but completely forgot! Fixed!

**Review by dani-garcia — APPROVED** [review]

**Pull request merged**